great_expectations/datasource/data_connector/inferred_asset_sql_data_connector.py

```diff
@@ -296,3 +300,32 @@ def get_available_data_asset_names_and_types(self) -> List[Tuple[str, str]]:
             A list of tuples consisting of available names and types
         """
         return [(asset["table_name"], asset["type"]) for asset in self.assets.values()]
+
+    def get_batch_definition_list_from_batch_request(self, batch_request: BatchRequest):
+        sub_cache: dict
+
+        self._validate_batch_request(batch_request=batch_request)
+
+        if len(self._data_references_cache) == 0:
+            self._refresh_data_references_cache()
+
+        batch_definition_list: List[BatchDefinition] = []
+        try:
+            sub_cache = self._data_references_cache[batch_request.data_asset_name]
+        except KeyError:
+            raise KeyError(
+                f"data_asset_name {batch_request.data_asset_name} is not recognized."
+            )
```

Comment:
This method was changed in `ConfiguredAssetSqlDataConnector` to accommodate requirements for adding `data_asset_name` as a `table_name` in a new `asset`, but that broke `InferredAssetSqlDataConnector` as it extends `Configured`. This override method ensures that using an altered `data_asset_name` that is not one of the inferred `assets` throws an error.